packages/@uppy/companion/src/server/provider/index.js

```diff
@@ -146,6 +146,8 @@ module.exports.addProviderOptions = (options, grantConfig) => {
       if (server.implicitPath) {
         // no url builder is used for this because grant internally adds the path
         grantConfig[authProvider].callback = `${server.implicitPath}${grantConfig[authProvider].callback}`
+      } else if (server.path) {
+        grantConfig[authProvider].callback = `${server.path}${grantConfig[authProvider].callback}`
```

Comment:
callback to an explicit `path` is already handled internally by [grant.js](https://github.com/simov/grant/) so there's no need to add this condition here. This is why we are passing the `path` into the config [here](https://github.com/transloadit/uppy/blob/master/packages/%40uppy/companion/src/server/provider/index.js#L123-L127).

Were you facing any issues with something that might be related to this behaviour? Otherwise I would say the existing implementation is fine as is.

Reply:
> Were you facing any issues with something that might be related to this behaviour? 
Yes, I did have, and resolve the issue by this PR

According to [grant.js' docs](https://github.com/simov/grant#path-prefix), 

> In case you want your callback routes prefixed, set them accordingly:

```javascript
{
  "facebook": {
    "callback": "/path/prefix/handle_facebook_callback"
  }
}
```

Reply:
@ifedapoolarewaju the bug blocks auth flow as usage indicated by grant.js

Reply:
@tranvansang you are right. It's just surprising that I had missed this all the while. Thank you so much for the PR.